accepted/00101-health-check-tool.md

```diff
@@ -0,0 +1,290 @@
+
+#### Uyuni Health Check
+- Create RPM packages in OBS/IBS.
+- Use container images building on OBS/IBS for the different components.
+- Provide an "standalone" version only (no integration with Monitoring stack). Via Uyuni server channel (or others so it is available for engineering or supporters).
+- Flag this as a Tech Preview.
+
+#### Saline
+- Research about using TCP instead of IPC sockets for Salt Master internal sockets.
+- Build Saline in OBS/IBS and provide it via RPM package for the Uyuni Server channel.
+- Saline runs inside "uyuni-server" container.
+
+### Phase 1 (after 5.0 GA):
+
+#### Uyuni Health Check
+- Enhance metrics, dashboards and alerts.
+- Allow user to enhance Loki pipeline to add filters and/or replace patterns to replace sensitive information.
+- Integration with Monitoring Stack. Enhance monitoring formula to enable Health Check monitoring.
+- Reuse configuration from "standalone" version.
+
+#### Saline
+- Provide "Saline" inside a separated container that attach to Salt Master TCP sockets from "uyuni-server" container.
+- Alternatively, explore sharing Salt Master IPC sockets between containers.
+
+[drawbacks]: #drawbacks
+
+
+* Resource Usage: Running additional services for log aggregation, analysis, and enhanced alerting could increase the resource demands on the server, including CPU, memory, and storage. This might necessitate upgrades to existing hardware or reevaluation of resource allocation in cloud environments to ensure optimal performance.
+
+* Security Considerations: With the introduction of new components for logging and alerting, there might be additional security considerations to address, including access controls, data encryption, and secure communication channels between services.
+
+* Alert Noise: Fine-tuning alert rules to balance sensitivity and specificity is critical. Without careful configuration, there's a risk of alert fatigue due to a high volume of non-critical alerts, potentially causing important alerts to be overlooked.
+
+
+# Alternatives
+
+[alternatives]: #alternatives
+
+- What other designs/options have been considered?
+
+  1. Running the Uyuni Health-Check-Tool along with the Uyuni Server container in the same pod. This could lead to a resource problem on the node running the pod.
+  2. Integration with current Monitoring stack. This has been considered in this RFC as an complementary integration for the standalone Health Check Tool.
+
+- What is the impact of not doing this?
+
+  1. Not having a Health Check Tools to monitor health of an Uyuni server. Poor monitoring when it comes to internal metrics, overall Uyuni server health, and particularly around Salt jobs and actions.
+
+
+# Unresolved questions
+
+[unresolved]: #unresolved-questions
+
+- What are the unknowns?
+
+Has Saline to be part of the current Uyuni Monitoring stack? As mentioned above, it does make sense to integrate Saline in the current Monitoring stack, probably as an "opt-in".
+
```

Comment:
From my perspective, Saline is primarily well-suited for larger customers. Although it could also offer advantages for smaller customers, I would suggest keeping it separate, perhaps as a distinct formula, to maintain clarity and tailored functionality for Saline.